### PR TITLE
upgrade debezium version to 1.9.2 for mysql source

### LINKED ISSUE #14425: Upgrade debezium version for MySQL source
We need to upgrade debezium version to 1.9.2 for mysql connector

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://github.com/airbytehq/airbyte/issues/14425

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSourceOperations.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public void setJsonField(final ResultSet resultSet, final int colIndex, final Ob
    */
   @Override
   protected void putBoolean(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index) throws SQLException {
-    node.put(columnName, resultSet.getInt(index) == 1);
+    node.put(columnName, resultSet.getInt(index) > 0);
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSourceOperations.java:110]: This is not related to debezium version upgrade, I found a bug while testing in Full refresh mode so went ahead and fixed it

File: airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java [hunk 2]
```diff
@@ -40,24 +39,9 @@ public void configure(final Properties props) {}
   public void converterFor(final RelationalColumn field, final ConverterRegistration<SchemaBuilder> registration) {
     if (Arrays.stream(DATE_TYPES).anyMatch(s -> s.equalsIgnoreCase(field.typeName()))) {
       registerDate(field, registration);
-    } else if (Arrays.stream(TEXT_TYPES).anyMatch(s -> s.equalsIgnoreCase(field.typeName()))) {
```
Comment (subodh1810) [inline airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java:43]: The new debezium version handles string values as expected so our custom logic is no required anymore

File: airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/FilteredFileDatabaseHistory.java [hunk 3]
```diff
@@ -1,149 +0,0 @@
...
- * created this class. In the method {@link #storeRecord(HistoryRecord)}, we introduced a check to
- * make sure only those records are being saved whose database name matches the database Airbyte is
- * syncing. We tell debezium to use this class by passing it as property in debezium engine. Look
- * for "database.history" property in {@link DebeziumRecordPublisher#getDebeziumProperties()}
- * Ideally {@link FilteredFileDatabaseHistory} should have extended
- * {@link io.debezium.relational.history.FileDatabaseHistory} and overridden the
- * {@link #storeRecord(HistoryRecord)} method but {@link FilteredFileDatabaseHistory} is a final
- * class and can not be inherited
- */
-public class FilteredFileDatabaseHistory extends AbstractDatabaseHistory {
```
Comment (subodh1810) [inline airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/FilteredFileDatabaseHistory.java:36]: The new debezium version filters out the schema history records only fir the database that its syncing so our custom logic is not required anymore

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java [hunk 4]
```diff
@@ -178,13 +177,16 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
     final JsonNode sourceConfig = database.getSourceConfig();
     if (isCdc(sourceConfig) && shouldUseCDC(catalog)) {
       final AirbyteDebeziumHandler handler =
-          new AirbyteDebeziumHandler(sourceConfig, MySqlCdcTargetPosition.targetPosition(database), MySqlCdcProperties.getDebeziumProperties(),
-              catalog, true);
+          new AirbyteDebeziumHandler(sourceConfig, MySqlCdcTargetPosition.targetPosition(database), true);
 
       final Optional<CdcState> cdcState = Optional.ofNullable(stateManager.getCdcStateManager().getCdcState());
       final MySqlCdcSavedInfoFetcher fetcher = new MySqlCdcSavedInfoFetcher(cdcState.orElse(null));
-      cdcState.ifPresent(cdc -> checkBinlog(cdc.getState(), database));
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java:186]: This is no longer required cause now if the last saved binlog position in database is purged, we would handle it gracefully and sync data from beginning

**Comment by grishick** [comment]
I cannot tell from the code if/how this change will break jobs that were created with the current version of MySQL Connector. Could you test that manually?

**Comment by subodh1810** [comment]
@grishick I tested this manually by setting up a mysql connector on version 0.5.15 (current version) in CDC mode and then ran few syncs and then switched the connector to use my local dev version which contained the changes from the PR and it worked fine so this is not going to cause a breaking change. 

I incorrectly assumed that it would have caused a breaking change cause of failing tests when I had upgraded the debezium version and realised that the `FileDatabaseHistory` is no longer required with the latest debezium version.

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlCdcStateHandler.java [hunk 5]
```diff
@@ -51,4 +51,9 @@ public AirbyteMessage saveState(final Map<String, String> offset, final String d
     return new AirbyteMessage().withType(Type.STATE).withState(stateMessage);
   }
 
+  @Override
+  public AirbyteMessage saveStateAfterCompletionOfSnapshotOfNewStreams() {
+    throw new RuntimeException("Snapshot of individual tables currently not supported in MySQL");
```
Comment (edgao) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlCdcStateHandler.java:56]: doublechecking: this only runs when someone adds new streams to an existing catalog, which isn't currently supported in the UI anyway?

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+  }
+
+  private void setupForComparisonBetweenFullRefreshAndCDCSnapshot(ImmutableList<JsonNode> data) {
+    executeQuery("CREATE DATABASE " + "test_schema" + ";");
+    executeQuery(String.format(
+        "CREATE TABLE %s.%s(%s INTEGER, %s Boolean, %s TINYINT(1), %s TINYINT(2), PRIMARY KEY (%s));",
+        "test_schema", "table_with_tiny_int", "id", "bool_col", "tiny_int_one_col",
+        "tiny_int_two_col", "id"));
+
+    executeQuery(String
```
Comment (edgao) [inline airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java:185]: nitpick: are these two queries identical? if yes, probably better to loop over `data` rather than copy/pasting them?

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+    streamWithoutPK.setSourceDefinedPrimaryKey(Collections.emptyList());
+    streamWithoutPK.setSupportedSyncModes(List.of(SyncMode.FULL_REFRESH));
+    addCdcMetadataColumns(streamWithoutPK);
+
+    streams.add(streamWithoutPK);
+    expectedCatalog.withStreams(streams);
+    return expectedCatalog;
+  }
+
+  //TODO : Enable this test once we fix handling of DATETIME values
```
Comment (edgao) [inline airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java:313]: which issue is this waiting for? https://github.com/airbytehq/airbyte/issues/14628 ?

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+    executeQuery(String
+        .format("INSERT INTO %s.%s (%s, %s) VALUES (%s, %s);", "test_schema",
+            "table_with_date_time",
+            "id", "datetime_col",
+            data.get(1).get("id").asInt(), data.get(1).get("datetime_col").asText()));
+    ((ObjectNode) config).put("database", "test_schema");
+  }
+
+  @Test
+  protected void syncShouldHandlePurgedLogsGracefully() throws Exception {
```
Comment (edgao) [inline airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java:386]: can this live in the base cdc source test? (and sources could explicitly opt out if it's not supported)

**Review by edgao — APPROVED** [review]
nice! just a couple nitpicks

**Review by grishick — APPROVED** [review]

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlCdcStateHandler.java (hunk 5 above)
Comment (subodh1810) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlCdcStateHandler.java:56]: @edgao correct plus the stream state type for mysql is LEGACY so even when the feature would be enabled in the UI, it would not be triggered for mysql

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java (hunk 7 above)
Comment (subodh1810) [reply airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java:309]: yup

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java (hunk 6 above)
Comment (subodh1810) [reply airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java]: done

**Comment by girarda** [comment]
This issue was closed because it was deprioritized. Feel free to reopen if needed.
